packages/medusa/src/utils/validators/is-nullable.ts

```diff
@@ -0,0 +1,29 @@
+import {
+  registerDecorator,
+  ValidationArguments,
+  ValidationOptions,
+} from "class-validator"
+import { IOptions } from "class-validator-jsonschema/src/options"
+
+export function IsNullable(validationOptions?: ValidationOptions) {
```

Comment:
suggestion: I think that we could instead use `@ValidateIf((object, value) => value !== null)` which could be wrapped here if we want to use `IsNullable`. It could looks like:

```ts
export function IsNullable(validationOptions?: ValidationOptions) {
  return ValidateIf((object, value) => value !== null, validationOptions)
}
```

Let me know wdyt :)

Reply:
I'm not sure I understand how this could this work. The main goal with the IsNullable is to be able to decorate a property that can accept `null` as an input value in order to be able to set `nullable: true` on its OAS representation. To do so, it must be registered as a class-validator decorator to participate in the [MetadataStorage](https://github.com/typestack/class-validator/blob/master/src/metadata/MetadataStorage.ts) discovery and registration.
```typescript
class Foo {
  IsOptional()
  IsNullable()
  IsString()
  foobar?: string | null
}  
```
Should convert to:
```json
"Foo": {
  "type": "object"
  "properties": {
    "foobar": {
      "type": "string",
      "nullable": true
    }
  } 
}
```

Reply:
got it thanks for the explanation